Added JBA PE branches

xgui3783 opened the pull request:
feat: add JBA PE branches
feat: updated JBA PE leave parent attr

xgui3783 committed:
feat: add JBA PE branches
feat: updated JBA PE leave parent attr

xgui3783 commented:
@lzehl seems like I have no write access to hbp/openminds_sands, so I had to push to my own fork and raise PR

additionally, I cannot set you as reviewer. but can you review nontheless?

lzehl commented:
@xgui3783 thank you for the PR. Yes I'll review it. You indeed had to go the external contributors way. I can imagine though adding you to the developer list for SANDS you may get involved a lot I imagine ;)

xgui3783 commented:
thought I would imagine having an external contributor perspective might also be helpful.

Looking forward to the review. 

also something strikes me: would it be useful to have a visualizer for the various `hasParent` attributes?

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_cingulate-gyrus--frontal-part.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part",
```
special signs such as "," are typically replaced by "_" in most namings we use

xgui3783 replied:
~~I don't see `,` ? do you mean `--` ?~~

edit: I think I see what you mean. replace `,` with `_`, and space with `-`

lzehl replied:
The name of "JBA_cingulate-gyrus--frontal-part" is "cingulate-gyrus, frontal-part".
Typically we replace special signs with "_" in the "@id" and file names. E.g.  "JBA_cingulate-gyrus_frontal-part"

But: I realized that in all other cases I think in the JBA uses the convention "frontal ...", "dorsal ..." etc. 
Should we just rename this completely anyways? To "JBA_frontal-cingulate-gyrus"?

xgui3783 replied:
so `JBA_{position_quantifier}-{name}` ?

I can do that

lzehl replied:
Yes. I think this is done in nearly all cases (e.g., "lateral occipital cortex", "frontal pole", etc)

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_cingulate-gyrus--frontal-part.jsonld:
```diff
@@ -0,0 +1,17 @@
+        }
+    ],
+    "hasVersion": null,
+    "lookupLabel": "JBA_cingulate-gyrus--frontal-part",
```
special signs such as "," are typically replaced by "_" in most namings we use

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_overall.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_overall",
```
this seems not correct... do you mean "brain" ?

lzehl replied:
or Julich-Brain ?

xgui3783 replied:
This is an error.

It seems in our definition there is a brain region called `overall`, (which serves as the root structure). 

do you think it's suitable to remove this instance, and have its children have no parents? 

(taken out of context the last line reads a little dark)

lzehl replied:
I would rename it to JBA_brain and use it as root structure for all GapMaps as well as telencephalon, metencephalon and diencephalon.

xgui3783 replied:
I doubled checked (https://gin.g-node.org/FZJ-INM1-BDA/20210615-julichbrain-v2.9.0/src/master/workingdir/hbp-d000006_julichbrain-cytoatlas_dev/JulichBrainCytoAtlas/julichbrain-ontology_29.0.json#L1728-L1740)

hartmut seems to use (nested) overal as super structure for the gapmaps (only) and nothing else. (In the json hierarchy there, it was suppose to be a child of cerebral cortex. Why in this case we have overall being parent of itself is a glitch (I use region name to generate id, and output jsonld name. so two regions with the same name meant one will overwrite the other)

I can certainly make adjustments, but I would like a few clarifications:

- I can certainly make a root region called `Julich-Brain` (independent of this `overall` branch node)
- how (if we should at all) change the branch nodes named `overall` (see screengrab below. or you could explore it at https://atlases.ebrains.eu/viewer/#/a:juelich:iav:atlas:v1.0.0:1/t:minds:core:referencespace:v1.0.0:dafcffc5-4826-4bf1-8ff6-46b8a31ff8e2/p:minds:core:parcellationatlas:v1.0.0:94c1125b-b87e-45e4-901c-00daee7f2579-290/@:0.0.0.-W000.._eCwg.2-FUe3._-s_W.2_evlu..7LIx..0.0.0..1LSm)
![Screenshot from 2021-12-09 17-33-59](https://user-images.githubusercontent.com/19381783/145437401-757fd574-6066-4f37-b286-56367f1745dc.png)

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_overall.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@type": "https://openminds.ebrains.eu/sands/ParcellationEntity",
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_overall"
```
cannot be the parent of itself. if it is the highest parent it should just have no parent

xgui3783 replied:
related to https://github.com/HumanBrainProject/openMINDS_SANDS/pull/151#discussion_r765961546

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_temporal-insula.jsonld:
```diff
@@ -0,0 +1,17 @@
+    ],
+    "hasVersion": null,
+    "lookupLabel": "JBA_temporal-insula",
+    "name": "temporal-insula",
```
I think the name should be without "-", no?

xgui3783 replied:
It is as the name in the json hartmut gave us

https://gin.g-node.org/FZJ-INM1-BDA/20210615-julichbrain-v2.9.0/src/master/workingdir/hbp-d000006_julichbrain-cytoatlas_dev/JulichBrainCytoAtlas/julichbrain-ontology_29.0.json

(require authentication/authorization, what is your gin user name? I will add you?)

we can certainly change it, just say the word.

lzehl replied:
well if this is what he defines leave it for now. we can discuss refinements with the Julich-Brain team later.

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-25.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
the "--" should be a "_" (see comment for parent)

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-33.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
the "--" should be a "_" (see comment for parent)

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op5.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-Op5",
```
this should be OP5 (as it was OP1 - OP4)

xgui3783 replied:
this also came directly from the json file hartmut passed us:

https://gin.g-node.org/FZJ-INM1-BDA/20210615-julichbrain-v2.9.0/src/master/workingdir/hbp-d000006_julichbrain-cytoatlas_dev/JulichBrainCytoAtlas/julichbrain-ontology_29.0.json#L1149

we can make the change now, if you give the OK.

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op5.jsonld:
```diff
@@ -0,0 +1,17 @@
+        }
+    ],
+    "hasVersion": null,
+    "lookupLabel": "JBA_Area-Op5",
```
this should be OP5 (as it was OP1 - OP4)

xgui3783 replied:
please see https://github.com/HumanBrainProject/openMINDS_SANDS/pull/151#discussion_r765939025

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op5.jsonld:
```diff
@@ -0,0 +1,17 @@
+    ],
+    "hasVersion": null,
+    "lookupLabel": "JBA_Area-Op5",
+    "name": "Area Op5 (Frontal Operculum)",
```
this should be OP5 (as it was OP1 - OP4)

xgui3783 replied:
please see https://github.com/HumanBrainProject/openMINDS_SANDS/pull/151#discussion_r765939025

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op6.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-Op6",
```
this should be OP6 (as it was OP1 - OP4) also below for lookupLabel and name

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op7.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-Op7",
```
this should be OP7 (as it was OP1 - OP4) also below for lookupLabel and name

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op8.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-Op8",
```
this should be OP8 (as it was OP1 - OP4) also below for lookupLabel and name

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-Op9.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-Op9",
```
this should be OP9 (as it was OP1 - OP4) also below for lookupLabel and name

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-0.jsonld:
```diff
@@ -2,9 +2,13 @@
         "@vocab": "https://openminds.ebrains.eu/vocab/"
     },
-    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1.0",
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1-0",
```
you can leave the "." in the "@id" this is fine.

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-0.jsonld:
```diff
@@ -2,9 +2,13 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Heschl-s-gyrus"
```
I would delete the "-" between Heschl and s

xgui3783 replied:
so a general rule is to remove all `'`s ?

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-0.jsonld:
```diff
@@ -64,7 +68,7 @@
         }
     ],
-    "lookupLabel": "JBA_Area-TE-1.0",
+    "lookupLabel": "JBA_Area-TE-1-0",
```
also here: the "." is okay

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-1.jsonld:
```diff
@@ -2,9 +2,13 @@
         "@vocab": "https://openminds.ebrains.eu/vocab/"
     },
-    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1.1",
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1-1",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-1.jsonld:
```diff
@@ -64,7 +68,7 @@
         }
     ],
-    "lookupLabel": "JBA_Area-TE-1.1",
+    "lookupLabel": "JBA_Area-TE-1-1",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-2.jsonld:
```diff
@@ -2,9 +2,13 @@
         "@vocab": "https://openminds.ebrains.eu/vocab/"
     },
-    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1.2",
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-1-2",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-1-2.jsonld:
```diff
@@ -64,7 +68,7 @@
         }
     ],
-    "lookupLabel": "JBA_Area-TE-1.2",
+    "lookupLabel": "JBA_Area-TE-1-2",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-2-1.jsonld:
```diff
@@ -2,9 +2,13 @@
         "@vocab": "https://openminds.ebrains.eu/vocab/"
     },
-    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-2.1",
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-2-1",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-2-1.jsonld:
```diff
@@ -49,7 +53,7 @@
         }
     ],
-    "lookupLabel": "JBA_Area-TE-2.1",
+    "lookupLabel": "JBA_Area-TE-2-1",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-2-2.jsonld:
```diff
@@ -2,9 +2,13 @@
         "@vocab": "https://openminds.ebrains.eu/vocab/"
     },
-    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-2.2",
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Area-TE-2-2",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-TE-2-2.jsonld:
```diff
@@ -49,7 +53,7 @@
         }
     ],
-    "lookupLabel": "JBA_Area-TE-2.2",
+    "lookupLabel": "JBA_Area-TE-2-2",
```
leave the "."

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-p24ab.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
"--" should be "_" see parent comment

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-p24c.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
"--" should be "_" see parent comment

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-p32.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
"--" should be "_" see parent comment

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-s24.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
"--" should be "_" see parent comment

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Area-s32.jsonld:
```diff
@@ -4,7 +4,11 @@
-    "hasParent": null,
+    "hasParent": [
+        {
+            "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_cingulate-gyrus--frontal-part"
```
"--" should be "_" see parent comment

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_Heschl-s-gyrus.jsonld:
```diff
@@ -0,0 +1,17 @@
+    "@context": {
+        "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_Heschl-s-gyrus",
```
I would use "JBA_Heschls-gyrus" (but this is up to you)

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_metencephalon.jsonld:
```diff
@@ -0,0 +1,13 @@
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_metencephalon",
+    "@type": "https://openminds.ebrains.eu/sands/ParcellationEntity",
+    "hasParent": null,
```
missing parent

xgui3783 replied:
In our hierarchy, metencephalon (& telencephalon) are root structures. 

What would the parents be?

lzehl replied:
as stated in the general comment: The brain is the root structure. Therefore I would rename JBA_overall to JBA_brain and then group all GabMaps and telencephalon, metencephalon and diencephalon below it.

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_telencephalon.jsonld:
```diff
@@ -0,0 +1,13 @@
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_telencephalon",
+    "@type": "https://openminds.ebrains.eu/sands/ParcellationEntity",
+    "hasParent": null,
```
missing parent

lzehl commented on instances/atlas/parcellationEntity/JBA/JBA_diencephalon.jsonld:
```diff
@@ -0,0 +1,13 @@
+    },
+    "@id": "https://openminds.ebrains.eu/instances/parcellationEntity/JBA_diencephalon",
+    "@type": "https://openminds.ebrains.eu/sands/ParcellationEntity",
+    "hasParent": null,
```
missing parent

lzehl reviewed (changes requested):
@xgui3783 thank you again for this PR and the support. I've added review comments (feel free to get back to me if there are questions or you need to discuss something) 

Also here some general comments/issues:

All jsonld miss an additional line at the end of each file (this is a something that github complains about and is an "error" in json.dump function in python...)

This is just a note for future cases. I would leave it here as it is, but typically we use a lowerCamelCase for the "@id" and the lookup labels. I did not do this as well for the Areas, because it would have messed up the correct writing of some of the labels. 

"JBA_overall" should be renamed to "JBA_brain" and should also be the parent of "telencephalon", "diencephalon" and "metencephalon".

xgui3783 commented:
discussed with @lzehl , here is what we decided:
- remove `overall`'s
- move all gapmaps under cerebral cortex
- capitalize all `Op` -> `OP`
- add Julich Brain as root structure

xgui3783 committed:
fix: new line at end of file
fix: do not escape dot in label name
fix: frontal cingulate gyrus
fix: gapmap parent
fix: root region

xgui3783 commented:
@lzehl can you take a look at the PR again?

For the root structure, I added https://github.com/xgui3783/SANDS/blob/feat_addJulichBrainPE/instances/atlas/parcellationEntity/JBA/JBA_julich-brain.jsonld

and I left https://github.com/HumanBrainProject/openMINDS_SANDS/pull/151#discussion_r765636714 unresolved, not sure if we would like to record somewhere to discuss naming of `temporal-insula`

Commit:
Update and rename JBA_julich-brain.jsonld to JBA_Julich-Brain.jsonld

Commit:
Update and rename JBA_Julich-Brain.jsonld to JBA_julich-brain.jsonld

lzehl commented:
@xgui3783 a big big thank you for this one ;)
